src/autoscaler/metricscollector/cmd/metricscollector/main.go

```diff
@@ -159,6 +176,65 @@ func main() {
 	logger.Info("exited")
 }
 
+func initDBLockRunner(conf *config.Config, logger lager.Logger, lockDB db.LockDB) ifrit.Runner {
+	dbLockMaintainer := ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
+		ttl := conf.DBLock.LockTTL
+		lockTicker := time.NewTicker(ttl)
+		readyFlag := true
+		owner := getOwner(logger, conf)
```

Comment:
If it is not deployed as a cf app,  we can use IP address and process ID, instead of using the one in config

If the component is deployed as one instance of a cf app, this will have issue.